.github/workflows/reusable/build-and-test.yml

```diff
@@ -1,25 +1,23 @@
-name: CI
+name: Build, lint and test
```

Comment:
procedural nitpick: nice to have refactors/moves separated out into their own commit so that they're not mingled with changes, makes for happier review + bisecting. No need to change for this PR!

Reply:
Noted, however for this particular file was changed into a reusable workflow so that it could be used by the other workflows in this PR. Not sure that counts as a standalone refactor?

Reply:
Ah fair, even if it's a bit of a change I like to keep moves in their own "logical chunk" but might have overlooked whether this specific change was actually suitable for that!